### ISSUE TITLE
Remove aria-live="polite" when slider is not focused

### ISSUE THREAD

**Issue by cshold** [kenwheeler/slick#1937]
`.slick-list` has `aria-live="polite"` on it, which tells screen readers to speak the content inside whenever it changes. This is great for the most part, but not ideal when you're not interacting with the slider. 

For example, imagine a screen reader scrolling down the page, past the slider, and still having the slide content read to them because `aria-live` exists on the element. It would be confusing to say the least.

I propose removing the aria attribute unless an element inside the slider is focused.

**Comment by laneystroup** [comment]
I second the need for this; it is very annoying to have your actions constantly interrupted by the carousel if you are a screen reader user.

**Comment by devben** [comment]
+1 
agreed this is very annoying for anyone using a screenreader

**Comment by laneystroup** [comment]
FYI, I have arrived at a workaround that I am pretty happy with. If you disable autoPlay and then set initialSlide to a random number, then you avoid the problem of the carousel interrupting the screen readers, you still some the variety for revisits, and the users can cycle through the other pictures if they like. The tradeoff of course, is that if you really want the users to see the other slides then they may not see those. However, it is probably better to place that information elsewhere on the page anyway rather than hiding it inside a carousel. For my purposes, it works perfectly, as the slides are more of a background feature (just rotating pictures).

To implement this, just set initialSlide to "Math.floor(Math.random() * (n + 1))", where n is the number of slides.

Hope that helps!

**Comment by depiction** [comment]
I agree that aria-live should be removed.

I tried the following to remove the attribute, then re-add it when the list has focus. Once focused on the slick list I waited until all slides had been rotated through. The attribute was added and removed appropriately, but the screen reader I use (VoiceOver) didn't read each slide as it changed. Since the live region isn't registered when added via JS the live-region can't be re-enabled on focus.

`			$slickList.removeAttr('aria-live');
			
			$slickList.on('focus', function() {
				$slickList.attr('aria-live', 'polite');
			});
			$slickList.on('blur').removeAttr('aria-live');`

**Comment by dotherightthing** [comment]
+1 My client has four (!) Slick sliders on the page, so things are very noisy for us here. ;-)

**Comment by edisch** [comment]
+1 On Chrome with VO for Mac, as a user tabs through each slide they receive audio feedback telling them the alt attribute of the slide image. However, on Firefox VO is not reading out the slides as they come through.

**Comment by JordanDDisch** [comment]
PR to remove aria live #2843

**Comment by classaxe** [comment]
Here's my fix for version 1.5.8:
This enables the `aria-live="polite"` attribute only when the container or any of its 'slides' has the focus.
Works best if you _DON'T_ disable the slide action on focus, but only on mouseover.

Old code:
```javascript
        _.$list = _.$slideTrack.wrap(
            '<div aria-live="polite" class="slick-list"/>').parent();
```
New code:
```javascript
        _.$list = _.$slideTrack.wrap(
            '<div class="slick-list" />').parent();
        $('.slick-list')
            .focusin(function() {
                $(this).attr('aria-live', 'polite');
            })
            .focusout(function() {
                $(this).removeAttr('aria-live');
            });
```
Then I reminified the code and it works for us properly now.

**Comment by joe-watkins** [comment]
👍 for total removal of `aria-live` on the parent, configuration option to kill it.

The W3C has [recommendations on accessible carousels](https://www.w3.org/WAI/tutorials/carousels/) which comes pretty close and I'd use that to help drive structure to support a wider audience. Notice the lack of fancy `aria` roles trying to force inappropriate `tab` or `listbox` roles. `aria-live` is only used to indicate which slide is active (I'd argue is overkill).. also super clean semantic markup is present :)

**Comment by cshold** [comment]
Closed by https://github.com/kenwheeler/slick/pull/2843
